modules/ssl/ssl_engine_init.c

```diff
@@ -574,8 +698,20 @@ static apr_status_t ssl_init_ctx_tls_extensions(server_rec *s,
      * protocol version(s) according to the selected (name-based-)vhost, which
      * is not possible at the SNI callback stage (due to OpenSSL internals).
      */
+#ifdef HAVE_OPENSSL_ECH
+    SSLSrvConfigRec *sc = mySrvConfig(s);
+    if (sc!=NULL && sc->echkeydir!=NULL) {
+        SSL_CTX_ech_set_callback(mctx->ssl_ctx, ssl_callback_ECH);
+    } else {
+        ap_log_error(APLOG_MARK, APLOG_TRACE4, 0, s, APLOGNO(10530)
```

Comment:
For `APLOG_TRACE?` logging we don't use `APLOGNO`s.

```suggestion
        ap_log_error(APLOG_MARK, APLOG_TRACE4, 0, s,
```

Reply:
also fixed that